src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -883,6 +885,82 @@ private function addCacheSection(ArrayNodeDefinition $rootNode)
         ;
     }
 
+    private function addOauthSection(ArrayNodeDefinition $rootNode)
+    {
+        $rootNode
+            ->children()
+                ->arrayNode('oauth')
+                    ->info('Oauth security provider configuration')
+                    ->{!class_exists(FullStack::class) && class_exists(GenericProvider::class) ? 'canBeDisabled' : 'canBeEnabled'}()
+                    ->fixXmlConfig('providers')
+                    ->children()
+                        ->arrayNode('default_options')
+                            ->children()
+                                ->enumNode('type')
+                                    ->info('The type of Oauth client needed: authorization_code, implicit, client, resource_owner')
+                                    ->values(['authorization_code', 'implicit', 'client', 'resource_owner'])
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('client_id')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('client_secret')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('authorization_url')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('redirect_uri')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('access_token_url')
+                                    ->isRequired()
+                                ->end()
```

Comment:
imo most of these options shouldn't have a "default", as for a client everything should be specific (e.g I shouldn't have the same key for multiple providers).

Especially as all of these (or almost) are required for each providers

Reply:
Thanks, I'm gonna change this 🙂